src/ert/config/summary_config.py

```diff
@@ -42,6 +46,7 @@ def read_from_file(self, run_path: str, iens: int) -> xr.Dataset:
         c_time = summary.alloc_time_vector(True)
         time_map = [t.datetime() for t in c_time]
         if self.refcase:
+            assert isinstance(self.refcase, set)
```

Comment:
I'm confused. `self.refcase` may be one of `None`, `Set[datetime]` and `List[str]`. But here we _assert_ that it must only be `set` if `bool(self.refcase) is True`. Are non-zero-sized `List[str]`s not possible?

Reply:
Will be happy to discuss alternatives for this, but `List[str]` is for deserializing it from json, and the conversion happens in `__post_init__`. Could use a pydantic class for it, but would add some overhead, but would make it easier to read.